notes/zrq/20210127-03-iris-oauth-proxy.txt

```diff
@@ -0,0 +1,428 @@
+         - --profile-url=${OAUTH2_SERVER_USER:?}
+         - --oidc-jwks-url=${OAUTH2_SERVER_JWKS:?}
+         - --cookie-secret=${OAUTH2_COOKIE_SECRET:?}
+         - --ssl-insecure-skip-verify
+         - --ssl-upstream-insecure-skip-verify
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+ labels:
+   k8s-app: ${deployname:?}-oauth-proxy
+ name: ${deployname:?}-oauth-proxy
+ namespace: default
+spec:
+ ports:
+ - name: http
+   port: 4180
+   protocol: TCP
+   targetPort: 4180
+ selector:
+   k8s-app: ${deployname:?}-oauth-proxy
+EOF
+
+
+# -----------------------------------------------------
+# Deploy the OAuth proxy.
+#[user@kubernator]
+
+    kubectl create \
+        --filename /tmp/${deployname:?}-oauth-proxy.yaml
+
+    >   deployment.apps/iris-oauth-proxy created
+    >   service/iris-oauth-proxy created
+
+
+# -----------------------------------------------------
+# Configure the oauth_proxy Ingress.
+#[user@kubernator]
+
+    #
+    # WARNING The auth-url and auth-signin URLs contain '$' values.
+    # WARNING If bash tries to fill them in, they will end up blank.
+    #   https:///oauth2/auth
+    #   https:///oauth2/start?rd
+    # WARNING This disables the authentication, leaving the protected resource exposed.
+    #
+
+    cat > /tmp/${deployname:?}-oauth-ingress.yaml << EOF
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+ annotations:
+   nginx.ingress.kubernetes.io/auth-url:    "https://\$host/${deployauthpath:?}/auth"
+   nginx.ingress.kubernetes.io/auth-signin: "https://\$host/${deployauthpath:?}/start?rd=\$escaped_request_uri"
+   nginx.ingress.kubernetes.io/auth-response-headers: "x-auth-request-user, x-auth-request-email"
+ name: ${deployname:?}-oauth-protected
+ namespace: default
+spec:
+ rules:
+ - host: ${deployhostname:?}
+   http:
+     paths:
+     - path: /
+       backend:
+         serviceName: http-svc
+         servicePort: 80
+ tls:
+ - hosts:
+   - ${deployhostname:?}
+   secretName: ${deployname:?}-tls-secret
+
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+ name: ${deployname:?}-oauth-protector
+ namespace: default
+spec:
+ rules:
+ - host: ${deployhostname:?}
+   http:
+     paths:
+     - path: /${deployauthpath:?}
+       backend:
+         serviceName: ${deployname:?}-oauth-proxy
+         servicePort: 4180
+ tls:
+ - hosts:
+   - ${deployhostname:?}
+   secretName: ${deployname:?}-tls-secret
+EOF
+
+
+# -----------------------------------------------------
+# Deploy the OAuth Ingress connectors.
+#[user@kubernator]
+
+    kubectl apply \
+        --filename /tmp/${deployname:?}-oauth-ingress.yaml
+
+    >   ingress.networking.k8s.io/iris-oauth-protected created
+    >   ingress.networking.k8s.io/iris-oauth-protector created
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+# Test the deployment.
+#[user@desktop]
+
+    firefox "http://claire.metagrid.xyz/frog" &
```

Comment:
Does this redirect to the Auth page? I assume the auth flow works successfully?

Reply:
Yes, the proxy redirects everything if the authenticated header is not set.

Reply:
Yep, the auth flow works.